### PR TITLE
Update dependencies & npm scripts

### PR DESCRIPTION
**Pull request by Robertmw**
- [x] Update dependencies versions;
- [x] Fix npm warnings about npm scripts & peer dependencies;
- [x] Replace `react-native` with `metro-react-native-babel-preset`.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,4604 @@
+{
```
Comment (ljharb) [inline package-lock.json]: Please remove this file; only apps should have lockfiles.

File: package.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   "description": "Sensible promise handling and middleware for redux",
   "main": "lib/index.js",
   "scripts": {
-    "prepublish": "npm run build",
+    "prepare": "npm run build",
```
Comment (ljharb) [inline package.json]: please keep this as “prepublish”

**Review by ljharb — CHANGES_REQUESTED** [review]
Given that the only change here that would require publishing is the Babel preset (the deps are already included by the semver range) I’m a bit confused about the urgency here. What’s the change in the Babel preset solving?

**Comment by Robertmw** [comment]
Hey, thanks for the review, I will make the changes today.

And sorry if I stress you or rush you, i don't mean to. I use this library in a project that is strict about dependency vulnerabilities (`npm audit`) hence the updated NPM package I made... And I cannot use the package as git dependency due to some project issues.

**Comment by ljharb** [comment]
@Robertmw in that case you can update the transitive dependencies in your lockfile using `npm audit fix` with no downstream changes required.

**Comment by ljharb** [comment]
Is `npm test` passing locally for you?

**Comment by Robertmw** [comment]
Fixed the `npm test` as well. I also added `package-lock.json` in `.gitignore`

File: .gitignore [hunk 3]
```diff
@@ -38,3 +38,5 @@ jspm_packages
 
 .DS_Store
 lib/
+
+package-lock.json
```
Comment (ljharb) [inline .gitignore:42]: let’s also add `package-lock=false` to `.npmrc`

File: .gitignore [hunk 4]
```diff
@@ -38,5 +38,3 @@ jspm_packages
 
 .DS_Store
 lib/
-
-package-lock.json
```
Comment (ljharb) [inline .gitignore]: to clarify; it should be here also.

**Comment by Robertmw** [comment]
This was the hardest PR in 2019 man.. 🤣

**Pull request closed without merging**